Library/Homebrew/cask/lib/hbc/cli/internal_stanza.rb

```diff
@@ -68,10 +65,17 @@ def run
             next
           end
 
-          if artifact_name && !value.key?(artifact_name)
-            opoo "no such stanza '#{artifact_name}' on Cask '#{cask}'" unless quiet?
-            puts ""
-            next
+          if value.nil? || (value.respond_to?(:to_a) && value.to_a.empty?) ||
+             (value.respond_to?(:to_s) && value.to_s == "{}") ||
+             (artifact_name && !value.include?(artifact_name))
+
+            if artifact_name
+              thing = artifact_name
+            else
+              thing = stanza
+            end
+
+            raise CaskError, "no such stanza '#{thing}' on Cask '#{cask}'"
           end
 
           if stanza == :artifacts
```

Comment:
The test error is coming from this part here, because `value` is not a `Hash` anymore, the `v` variable will always be nil.

Reply:
OK, I understand now that Sets have to be handled differently than Hashes.
Yes, I will rebase and hopefully my local brew will behave the same way as "Travis" brew.

Reply:
@reitermarkus YES! Rebase did the trick! Now finally my local brew seems to behave like "Travis brew". This will help fixing the final issues tremendously! Thank you for pointing out this blunder of mine.